WEEK1 STEP3 sonshinwoo

Sonshinwoo opened the pull request:
@yeahg-dev 
step3  초안입니다 ㅎㅎ 
소스트리랑 git 관련하여 공부하다보니 조금 늦었네요 ㅎㅎ 
step3에선 저번에 말씀 해주셨던 카르마 내용과 순서도 마지막 부분에 대해서 적용해봤습니다 ㅎ

Commit:
<implement> : week1 step3 초본

저번 피드백때 말씀해주신 카르마(커밋 템플릿??)에 대해 알아보고 적용해봤습니다.
다만 릴리님께서 링크를 보내주신건 영어를 기반으로 하는 내용이기에 구글에서 좀 찾아보고 한글식으로 적용된 템플릿을 적용해봤어요 ㅎ

yeahg-dev commented on week1 step3/step3/main.swift:
```diff
@@ -0,0 +1,48 @@
+
+import Foundation
+
+var countingPeperoHeight = Int(0)
```
Swift API가이드라인에는 아래와 같은 가이드가 있어요👀
>The names of other types, properties, variables, and constants should read as nouns.

변수는 명사형으로 이름 지어주세요:)

Sonshinwoo replied:
동명사 형태로 사용해도 된다고 생각 했는데 아니였군요 시간 날때마다 천천히 다시 읽어봐야겠어요 ㅎㅎ

Sonshinwoo replied:
git 명령어가 약간 꼬여서 실수로 순서도 사진을 날려버렸나보네요 ㅎㅎ 피드백 해주신 프린트 부분이랑 순서도를  수정해서 pr 하겠습니다 ㅎ

yeahg-dev commented on week1 step3/step3/main.swift:
```diff
@@ -0,0 +1,48 @@
+    makePeperoStickOf(height: stickheight)
+}
+
+func makePeperoBodyOf(type: String, topping: String) {
+    countingPeperoHeight = countingPeperoHeight + 1
+    if countingPeperoHeight % 2 == 1 {
+        print("\(topping)", terminator: "")
+    }else {
+        print(" ", terminator: "")
+    }
+    print("\(type)", terminator: "")
+    
+    if countingPeperoHeight % 2 == 1 {
+        print(" ")
+    }else {
+        print("\(topping)")
+    }
+}
```
실행상으로는 전혀 문제가 없지만, 코드의 내용을 이해하기 가 좀 어렵습니다.
문자열 보간법을 사용해서 코드가 하는일(코드가 어떤 것을 출력하고자하는지) 좀 더 직관적으로 표현하면 좋을 것 같아요~

그리고 같은 조건을 두 두번 검사하고 있네요. 이렇게 작성하신 이유가 있을까요?

Sonshinwoo replied:
제가 만들고도 나중에 봤을 때 이해가 잘 안된다고 생각해서 고쳐보려고 했는데 마땅히 떠오르는 아이디어가 없어 일단 제출했습니다. 릴리님이 말씀하신 문자열 보간법을 사용하면 좀 더 간단하게 표현해볼 수 있을거 같아요 ㅎ 피드백 내용 반영해서 다시 pr 할게요 ㅎ

Sonshinwoo replied:
그리고 같은 조건을 두 두번 검사하고 있네요. 이렇게 작성하신 이유가 있을까요?

A. :  바디타입을 기준으로 토핑을 왼쪽과 오른쪽으로 나눠서 진행하기 위해 두번을 반복해서 사용했는데요 사실 자세히 보시면 ㅎㅎ 조건은 같지만 if 문의 첫번쨰 조건과 else 조건이 반대로 만들어져 있습니다. 이렇게 제작한 이유는 왼쪽 토핑은 홀수번쨰에 입력값(*, #)이 들어가고 짝수번째엔 공백으로 표시가 되어 있고 오른쪽 토핑은 왼쪽과 반대로 홇수가 공백값이고 짝수가 토핑이 들어가 있습니다.  글을 쓰다보니 같은 조건문으로 하게되면 다른사람이 제 코드를 봤을 때 많이 혼동스러울 수도 있을꺼 같네요  마지막 if문에서 %2 == 1 값(홀수) 값이 아닌 % 2 ==(0)(짝수)로 작성하고 주석을 달아놨으면 좀 덜 햇갈려 보일 수 있을꺼 같은데 릴리님 생각은 어떠신가요??

yeahg-dev replied:
신우님께서 설명해주신 것 처럼 빼빼로는 현재 길이가 홀수 일 때와 짝수 일 때 두가지 경우에서 코드의 분기가 일어납니다. 
저의 개인적 생각으로는 두차례 나눠 작성한 홀수에 대한 코드와 짝수에 대한 코드를 하나의 조건문 안에서 합쳐주면, 각 조건일 때 하는 행동도 직관적으로 드러날 것 같아요!

yeahg-dev commented on week1 step3/step3/main.swift:
```diff
@@ -0,0 +1,48 @@
+var countingPeperoHeight = Int(0)
+
+func makePeperoOf(bodyheight: Int, topping: String, peperotype: String, stickheight: Int) {
+    for _ in 1 ... bodyheight {
+        makePeperoBodyOf(type: peperotype, topping: topping)
+    }
```
- 길이(`Int`)에 따른 막대(빼빼로 손잡이)를 그리는 함수
로 분리해볼까요?

Sonshinwoo replied:
@yeahg-dev 
릴리님이 태그주신 부분에는 빼빼로 손잡이를 그리는 함수가 포함되어 있지 않은데 말씀해주신 func makeicecreamof 함수에 포함되어 있는 13~15 코드 밑에  makePeperoStickOf(height: stickheight) 를 따로 분리해서 만들라는 말씀이신가요???

yeahg-dev replied:
아이코 제가 말실수를 했네요. 혼동을드려 죄송합니다. 길이에따른 막대를 그리는 함수가 아닌 길이에따른 빼빼로 몸통을 그리는 함수로 분리해보자는 말이었습니다.😅

yeahg-dev commented on week1 step3/step3/main.swift:
```diff
@@ -0,0 +1,48 @@
+
+
+makePeperoOf(bodyheight: 10, topping: " ", peperotype: "***", stickheight: 4)
+print() //출력값 띠어쓰기로 사용
```
주석을 사용하지 않고도 코드를 이해할 수 있도록 공백을 만들어주는 함수를 생성해보는 건 어떨까요?

Sonshinwoo replied:
makeicecreamOf 함수 내부에 terminator: "/n/n" 을 구현해 공백함수를 만들었습니다.

yeahg-dev reviewed (changes requested):
안녕하세요 신우님!

요구사항과 같이 빼빼로를 정확하게 그리는 함수를 잘 만들어주셨어요💯
그리고 전체적인 함수의 네이밍도 명확해서 이해하기 쉬웠습니다👍

컨벤션을 공부하시고, 커밋 타이틀도 컨벤션에 따라 잘 작성해주셧습니다😃
그런데 커밋의 본문에는 **해당 커밋과 관련된, 코드와 관련된** 추가적 설명이 들어가는 게 좋을 것 같아요~ 
예를 들면 코드를 왜 이렇게 작성했는지, 커밋의 목적은 무엇인지 와 같은 내용을 작성할 수 있을 것 같아요. 
개인적으로 저는 커밋의 본문을 커밋 타이틀만으로는 설명이 부족하다 싶을 때! 
해당 커밋에서 한 일들을 간략하게 정리하거나, 목적과 의도를 추가적으로 설명하거나 합니다.
이건 저의 스타일이니 참고만해주세요!

아직, 다음 스텝으로 넘어가기엔 요구사항을 덜 클리어한 부분이 있네요🥲

### 1. 파일 생성은 같은 프로젝트에 하기!
step3에 대한 프로젝트를 또 새로 생성하셨습니다. 
> Step 2에 작성했던 `Command Line Tool` 프로젝트 내부에 새로운 파일을 생성합니다.
기존 스텝을 진행하던 프로젝트 내에서 새로운 파일을 생성하도록 해주세요.

### 2. 실행 예시의 정보(길이, 몸통, 토핑, 막대길이)를 함께 출력합니다.

아래의 정보를 출력하는 코드가 아직 구현이 안되어있네요.
정보를 출력하는 함수를 만들어볼까요?
```swift
<정보>
길이: 10
몸통: ***
토핑: 
막대길이: 4
```
### 3. 순서도 확인
2주차에 해당하는 순서도가 올라와있네요..?
스텝3에 해당하는 순서도도 작성후 첨부바랍니다~!

위 요구사항 구현과 제가 남긴 코멘트에 대해서도 한번 다시 생각해보시고
재리뷰 요청주세요~!

Commit:
<Fix> : WEEK1 STEP3 피드백 관련내용 수정사항

말씀해주셨던 피드백 내용 수정사항이며 내용을 아래와 같습니다

1.countingpeperotype 명사로변경
동명사의 형태가 아닌 일반명사 형태로 바꿨습니다. 다만 단어 1개의 명사로만 표기하기엔 표현의 한계가 있어 명사 + of + 명사 의 형태로 바꿔봤습니다.

2. print의 정보구현(길이: ~ , 몸통: ~, 스틱길이:, 토핑: )
func makeicecreamof 함수에 print함수를 넣어 구현하였습니다. 처음엔 separator 와 terminator를 Print에 넣어 한줄로 구현하려 했으나 오히려 가독성이 더 떨어질꺼 같아 한줄씩 구현했습니다.

3. 순서도 사진 업로드
저번 피드백때 사진이 누락되어 재업로드 했습니다

4. 문자 보간법 사용
지적 해주셨던 func makeicecreambodyof 부분에 변수명을 내부에서 재설정하여 각 코드마다 다시 적용하여 직관성을 높였습니다.

5. step2에 코드 다시 올리기
step3 부분을 삭제 하고 기존에 제작했던 코드에 전코드를 삭제하고 step3 코드를 옮겨놨습니다.

모르는 부분
말씀해주셨던 13~15줄부분에 빼빼로 스틱부분을 분리 하라고 하신건 이해가 잘안돼서일단 그대로 놔두고 커밋했습니다.

Commit:
<Fix> : WEEK1 STEP3 피드백 관련내용 수정사항

말씀해주셨던 피드백 내용 수정사항이며 내용을 아래와 같습니다

1.countingpeperotype 명사로변경

동명사의 형태가 아닌 일반명사 형태로 바꿨습니다. 다만 단어 1개의 명사로만 표기하기엔 표현의 한계가 있어 명사 + of + 명사 의 형태로 바꿔봤습니다.

2. print의 정보구현(길이: ~ , 몸통: ~, 스틱길이:, 토핑: )

func makeicecreamof 함수에 print함수를 넣어 구현하였습니다. 처음엔 separator 와 terminator를 Print에 넣어 한줄로 구현하려 했으나 오히려 가독성이 더 떨어질꺼 같아 한줄씩 구현했습니다.

3. 순서도 사진 업로
드
저번 피드백때 사진이 누락되어 재업로드 했습니다

4. 문자 보간법 사용

지적 해주셨던 func makeicecreambodyof 부분에 변수명을 내부에서 재설정하여 각 코드마다 다시 적용하여 직관성을 높였습니다.

5. step2에 코드 다시 올리기
step3 부분을 삭제 하고 기존에 제작했던 코드에 전코드를 삭제하고 step3 코드를 옮겨놨습니다.

말씀해주셨던 13~15줄부분에 빼빼로 스틱부분을 분리 하라고 하신건 이해가 잘안돼서일단 그대로 놔두고 커밋했습니다.

Commit:
Merge branch 'ss_4_sonshinwoo' of https://github.com/Sonshinwoo/swift_starter-week1 into ss_4_sonshinwoo

Commit:
<Fix> : WEEK1 STEP3 피드백 관련내용 수정사항
말씀해주셨던 피드백 내용 수정사항이며 내용을 아래와 같습니다

1.countingpeperotype 명사로변경
동명사의 형태가 아닌 일반명사 형태로 바꿨습니다. 다만 단어 1개의 명사로만 표기하기엔 표현의 한계가 있어 명사 + of + 명사 의 형태로 바꿔봤습니다.

2. print의 정보구현(길이: ~ , 몸통: ~, 스틱길이:, 토핑: )
func makeicecreamof 함수에 print함수를 넣어 구현하였습니다. 처음엔 separator 와 terminator를 Print에 넣어 한줄로 구현하려 했으나 오히려 가독성이 더 떨어질꺼 같아 한줄씩 구현했습니다.

3. 순서도 사진 업로드
저번 피드백때 사진이 누락되어 재업로드 했습니다

4. 문자 보간법 사용
지적 해주셨던 func makeicecreambodyof 부분에 변수명을 내부에서 재설정하여 각 코드마다 다시 적용하여 직관성을 높였습니다.

5. step2에 코드 다시 올리기
step3 부분을 삭제 하고 기존에 제작했던 코드에 전코드를 삭제하고 step3 코드를 옮겨놨습니다.

모르거나 이해가 잘 안되는 부분

1.말씀해주셨던 13~15줄부분에 빼빼로 스틱부분을 분리 하라고 하신건 이해가 잘안돼서일단 그대로 놔두고 커밋했습니다.

2. 커밋 내용이나 띄어쓰기가 맘에들지 않아 git commit --amend를 사용해 고쳐서 다시 commit 과 Push를 했지만  충돌이 일어나 sourcetree도 멈추고 commit도 원격 저장소에 올라가 있지도 않았습니다.  git Commit Pull을 사용하여 다시 원상복구를 시켜놓긴 했는데 원래 --amend를 사용하며 이렇게 에러가 발생하나요??
